pkg/controller/constrainttemplate/constrainttemplate_controller.go

```diff
@@ -491,20 +492,35 @@ func (r *ReconcileConstraintTemplate) handleUpdate(
 			currentVap = nil
 		}
 		logger.Info("get vap", "vapName", vapName, "currentVap", currentVap)
-		newVap := &admissionregistrationv1beta1.ValidatingAdmissionPolicy{}
+		newVap := &unstructured.Unstructured{}
+		newVap.SetGroupVersionKind(schema.GroupVersionKind{Group: groupVersion.Group, Version: groupVersion.Version, Kind: "ValidatingAdmissionPolicy"})
 		transformedVap, err := transform.TemplateToPolicyDefinition(unversionedCT)
 		if err != nil {
-			logger.Info("transform to vap error", "vapName", vapName, "error", err)
+			logger.Error(err, "transform to vap error", "vapName", vapName)
 			createErr := &v1beta1.CreateCRDError{Code: ErrCreateCode, Message: err.Error()}
 			status.Status.Errors = append(status.Status.Errors, createErr)
 			err := r.reportErrorOnCTStatus(ctx, ErrCreateCode, "Could not transform to vap object", status, err)
 			return reconcile.Result{}, err
 		}
 		if currentVap == nil {
-			newVap = transformedVap.DeepCopy()
+			err = r.scheme.Convert(transformedVap, newVap, nil)
```

Comment:
Same concern about convert/unstructured. We can probably just use typed objects.

Reply:
Using typed conversion.